subprojects/packagefiles/google-benchmark/meson.build

```diff
@@ -28,13 +29,19 @@ benchmark_sources = [
 benchmark_main_sources = ['src/benchmark_main.cc']
 inc = include_directories('include')
 
-thread_dep = dependency('threads')
+
+lib_benchmark_deps = [ dependency('threads') ]
+
+if target_machine.system() == 'windows'
```

Comment:
The meson documentation describes host_machine as the environment that the built library/program runs on, and target_machine as the environment that the built program (running on host_machine) generates code for.

>  *target machine* is the machine on which the compiled binary's output will run, *only meaningful* if the program produces machine-specific output.

Reply:
Yes, cross-compilation is super confusing. https://mesonbuild.com/Cross-compilation.html

Reply:
Yeah, so I took this to mean that the target machine is what is relevant.  I.e. a hypothetical cross compile on a Linux machine to a windows binary would need the dep.  Similarly, a cross compile on a windows machine to a linux binary would not.  Am I wrong?

Reply:
I see that most everybody on github uses host_machine == 'windows' but I think everybody is basically wrong.  :-)  It just doesn't matter in practice because if either host or target is 'windows' then they both are (in practice).

Reply:
build_machine is typically used for helper programs which are built and then run as part of the build process to do other things.

host_machine is typically used for defining the machine you're cross-compiling for.

target_machine is basically used only when the project you are building is a cross-compiler, and you are e.g. cross-compiling from your build_machine laptop running linux, trying to cross-compile a host_machine Windows compiler, and then when you have the built compiler, you're going to distribute it to Windows users, who are going to use that compiler to cross-compile the target_machine macOS programs.

Reply:
Oh, I get it.  target machine is yet a third thing and host machine is what I thought target was!

Reply:
The terms used are very fun!

I like to ironically think of them like this:

build_machine is the machine you use as a buildbot host

host_machine is the machine you're building for.

target_machine is the machine that your target is targeting.

Also, this is basically the case with autotools too, but personally I don't trust any code I've written to be cross-compile safe unless I've double checked the manual right then to be sure.